File: backend/BYTUBE/Models/VideoModel.cs
namespace BYTUBE.Models
{
    public class VideoModel
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }

        public int Views { get; set; }

        public List<string> Tags { get; set; }

        public DateTime Created { get; set; }

        public string Duration { get; set; }

        public string PreviewUrl { get; set; }
        public string VideoUrl { get; set; }

        public ChannelModel Channel { get; set; }
    }
}

File: frontend/src/types/models/VideoModel.ts
import ChannelModel from "./ChannelModel";

enum Access {
  All = 0,
  LinkOnly,
  Private,
}

enum Status {
  NoLimit = 0,
  Limited,
  Blocked,
}

interface VideoModel {
  id: string;
  title: string;
  description: string;
  created: string;
  duration: string;

  forAdults: boolean;

  videoAccess: Access;
  videoStatus: Status;

  views: number;
  reportsCount?: number;

  tags?: string[];

  previewUrl: string;
  videoUrl?: string;

  channel?: ChannelModel;
}

export default VideoModel;

export { Status, Access };
